pkg/encoding/fix.go

```diff
@@ -47,3 +53,18 @@ func fixPersistentVolumes(app *spec.App) error {
 	}
 	return nil
 }
+
+func fixConfigData(app *spec.App) error {
+	// if only one configData is defined and it's name is not specified
+	if len(app.ConfigData) == 1 && app.ConfigData[0].Name == "" {
```

Comment:
How about doing this like the `fixServices` and `fixPersistentVolumes` functions in which first we loop over `app.*` and then we check if length is 1 or not, just for the sake of consistency, and makes it a tiny bit more readable.

Also, we won't increase the complexity by doing that anyway because if the length is one, then it's being looped over just once!

WDYT?